safe/utilities/test/test_upload_layer_geonode.py

```diff
@@ -0,0 +1,52 @@
+# coding=utf-8
+
+"""Test for uploading a layer to Geonode."""
+
+__copyright__ = "Copyright 2018, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+import unittest
+
+from safe.test.qgis_app import qgis_app
+from safe.test.utilities import load_test_vector_layer, load_test_raster_layer
+from safe.utilities.upload_layer_geonode import upload, login_user
+
+qgis_app()
+
+LOGIN = ''
+PASSWORD = ''
+GEONODE_URL = 'http://staging.geonode.kartoza.com'
+
+
+class GeonodeUploadTest(unittest.TestCase):
+
+    """Test Geonode upload.
+
+    WARNINGS:
+
+    Most of tests here are fragile as they rely on an external website.
+    Their HTML might change and break the test. These tests are disabled
+    because we use our own credentials.
+    """
+
+    def setUp(self):
+        self.raster = load_test_raster_layer(
+            'gisv4', 'hazard', 'earthquake.asc')
+        self.vector = load_test_vector_layer(
+            'exposure', 'airports.shp')
+
+    @unittest.skipUnless(LOGIN, 'You need to fill LOGIN and PASSWORD above.')
+    def test_login_and_upload_layers(self):
+        """Test login page."""
+        # Connexion
```

Comment:
Is it French? :D

Reply:
No at all: https://en.wikipedia.org/wiki/Connexion english wikipage ;-)